### PR TITLE
ddt removal in cli architecture and compute resource tests

### PR DESCRIPTION
**Pull request by sthirugn**
Also made:
* Made pylint 100%
* optimized/renamed test data

### CODE REVIEW

**Comment by sthirugn** [comment]
Test results:

test_architecture:
```sh
Finding files... done.
Importing test modules ... done.

----------------------------------------------------------------------
Ran 23 tests in 172.013s

OK

```

test_computeresource:
```sh
Finding files... done.
Importing test modules ... done.

----------------------------------------------------------------------
Ran 14 tests in 144.031s

OK (skipped=1)
```

**Comment by sthirugn** [comment]
this pr is intentionally created with 2 commits since two different modules are involved.

**Comment by sthirugn** [comment]
rebased

File: tests/foreman/cli/test_architecture.py [hunk 1]
```diff
@@ -33,27 +24,22 @@ def test_negative_update(self, data):
...
 
         """
-        try:
-            new_obj = make_architecture()
-        except CLIFactoryError as err:
-            self.fail(err)
+        for data in invalid_values_list():
+            with self.subTest(data):
+                try:
+                    new_obj = make_architecture()
```
Comment (omaciel) [inline tests/foreman/cli/test_architecture.py]: You can move the creation outside the loop and perform only the update inside the loop to speed up the test.

File: tests/foreman/cli/test_computeresource.py [hunk 2]
```diff
@@ -299,47 +334,42 @@ def test_update_positive(self, options):
...
-        )
-        self.assertEqual(
-            result.stdout['url'],
-            options.get('url', comp_res['url'])
-        )
-        self.assertEqual(
-            result.stdout['provider'].lower(), comp_res['provider'].lower())
+        for options in valid_update_data():
+            with self.subTest(options):
+                comp_res = make_compute_resource()
```
Comment (omaciel) [inline tests/foreman/cli/test_computeresource.py]: Move the object creation outside the loop and perform all updates inside the loop to speed up the test.

File: tests/foreman/cli/test_computeresource.py [hunk 3]
```diff
@@ -347,24 +377,26 @@ def test_update_negative(self, options):
...
-        self.assertEqual(result.stdout['name'], comp_res['name'])
-        options.pop('new-name', None)
-        for key in options.keys():
-            self.assertEqual(comp_res[key], result.stdout[key])
-
-    @data('true', 'false')
-    def test_set_console_password_v1(self, set_console_password):
+        for options in invalid_update_data():
+            with self.subTest(options):
+                comp_res = make_compute_resource()
```
Comment (omaciel) [inline tests/foreman/cli/test_computeresource.py]: Move the object creation outside the loop and perform all updates inside the loop to speed up the test.

File: tests/foreman/cli/test_computeresource.py [hunk 4]
```diff
@@ -395,18 +428,20 @@ def test_set_console_password_v2(self, set_console_password):
...
-        result = ComputeResource.update({
-            'name': name,
-            'set-console-password': set_console_password,
-        })
-        self.assertEqual(result.return_code, 0)
-        self.assertEqual(len(result.stderr), 0)
+        for set_console_password in ('true', 'false'):
+            with self.subTest(set_console_password):
+                name = gen_string('utf8')
+                result = ComputeResource.create({
```
Comment (omaciel) [inline tests/foreman/cli/test_computeresource.py]: Move the object creation outside the loop and perform all updates inside the loop to speed up the test.

**Comment by omaciel** [comment]
ACK pending comments meant to avoid calling object creation inside loop to speed up the test suite.

**Comment by elyezer** [comment]
ACK pending @omaciel's suggestion

File: tests/foreman/cli/test_computeresource.py (hunk 2 above)
Comment (sthirugn) [inline tests/foreman/cli/test_computeresource.py]: The function valid_update_data() supplies variety of test data like name, new-name, url, description.  By reusing compute resource, this test will look more complicated (I will have to store the old values in separate variables after every iteration). I like to leave it this way.

File: tests/foreman/cli/test_computeresource.py (hunk 3 above)
Comment (sthirugn) [inline tests/foreman/cli/test_computeresource.py]: same explanation as the above

File: tests/foreman/cli/test_architecture.py (hunk 1 above)
Comment (sthirugn) [inline tests/foreman/cli/test_architecture.py]: :+1:

File: tests/foreman/cli/test_computeresource.py (hunk 4 above)
Comment (sthirugn) [inline tests/foreman/cli/test_computeresource.py]: :+1:

**Comment by sthirugn** [comment]
test_architecture.py:

before:
```sh
Finding files... done.
Importing test modules ... done.

----------------------------------------------------------------------
Ran 23 tests in 182.679s

OK
```

after:
```sh
Finding files... done.
Importing test modules ... done.

----------------------------------------------------------------------
Ran 23 tests in 154.063s

OK
```

before:
```sh
Finding files... done.
Importing test modules ... done.

----------------------------------------------------------------------
Ran 14 tests in 153.073s

OK (skipped=1)
```

after:
```sh
Finding files... done.
Importing test modules ... done.

----------------------------------------------------------------------
Ran 14 tests in 137.032s

OK (skipped=1)
```

**Comment by sthirugn** [comment]
@omaciel Rebased and ready to review/merge

**Comment by omaciel** [comment]
@sthirugn ACK

**Comment by sthirugn** [comment]
Part of https://github.com/SatelliteQE/robottelo/issues/2769

**Pull request merged**